File: android/networking/src/main/kotlin/dev/example/moodlenative/networking/LmsModuleCompletionRequirement.kt
package dev.example.moodlenative.networking

data class LmsModuleCompletionRequirement(
    val id: String,
    val text: String,
    val isComplete: Boolean,
) {
    val symbolName: String
        get() = if (isComplete) "checkmark.circle.fill" else "checkmark.circle.dotted"

    val accessibilityLabel: String
        get() = "$text、${if (isComplete) "完了" else "未完了"}"
}

File: ios/Shared/Sources/Networking/LmsModuleCompletionRequirement.swift
import Foundation

public struct LmsModuleCompletionRequirement: Hashable, Identifiable, Sendable {
    public let id: String
    public let text: String
    public let isComplete: Bool

    public var symbolName: String {
        isComplete ? "checkmark.circle.fill" : "checkmark.circle.dotted"
    }

    public var accessibilityLabel: String {
        "\(text)、\(isComplete ? "完了" : "未完了")"
    }
}
